src/pymor/models/neural_network.py

```diff
@@ -202,6 +202,83 @@ def _compute_solution(self, mu=None, **kwargs):
+            to solution space. Should be an instance of
+            :class:`~pymor.models.neural_network.FullyConnectedNN` with input size that
+            matches the (total) number of parameters and output size equal to the
+            dimension of the reduced space.
+        parameters
+            |Parameters| of the reduced order model (the same as used in the full-order
+            model).
+        output_functional
+            |Operator| mapping a given solution to the model output. In many applications,
+            this will be a |Functional|, i.e. an |Operator| mapping to scalars.
+            This is not required, however.
+        products
+            A dict of inner product |Operators| defined on the discrete space the
+            problem is posed on. For each product with key `'x'` a corresponding
+            attribute `x_product`, as well as a norm method `x_norm` is added to
+            the model.
+        error_estimator
+            An error estimator for the problem. This can be any object with
+            an `estimate_error(U, mu, m)` method. If `error_estimator` is
+            not `None`, an `estimate_error(U, mu)` method is added to the
+            model which will call `error_estimator.estimate_error(U, mu, self)`.
+        visualizer
+            A visualizer for the problem. This can be any object with
+            a `visualize(U, m, ...)` method. If `visualizer`
+            is not `None`, a `visualize(U, *args, **kwargs)` method is added
+            to the model which forwards its arguments to the
+            visualizer's `visualize` method.
+        name
+            Name of the model.
+        """
+
+        def __init__(self, T, nt, neural_network, parameters={}, output_functional=None,
+                     products=None, error_estimator=None, visualizer=None, name=None):
+
+            super().__init__(products=products, error_estimator=error_estimator,
+                             visualizer=visualizer, name=name)
+
+            self.__auto_init(locals())
+            self.solution_space = NumpyVectorSpace(neural_network.output_dimension)
+            if output_functional is not None:
+                self.dim_output = output_functional.range.dim
+
+        def _compute_solution(self, mu=None, **kwargs):
+
+            U = self.solution_space.empty(reserve=self.nt)
+
+            # create tensor of parameters at different time instances
+            def time_dependent_parameter(t):
+                return [mu.get_time_dependent_value(param)(t) for param in mu]
+
+            parameters = torch.DoubleTensor([[time_dependent_parameter(t)
+                                              for t in np.linspace(0., self.T, self.nt)]])
+
+            # obtain (reduced) coordinates by forward pass of the parameter values
+            # through the neural network
+            result_neural_network = self.neural_network(parameters[..., 0]).data.numpy()
```

Comment:
Maybe I'm confused, but shouldn't be the time be the first index. So `[0]` instead of `[..., 0]`?

Reply:
This code also works for vector valued outputs, right?

Reply:
It should, yes. I will take another look.